File: ios/test/mappers/CartItemMapperTest.m
#import <XCTest/XCTest.h>
#import "CartItemMapper.h"

@interface CartItemMapperTest: XCTestCase

@end

@implementation CartItemMapperTest

- (void)test_toModel {
  NSDictionary *dictionary = @{
    @"itemId": @"testId",
    @"price": @12.34,
    @"quantity": @56.78
  };
  
  EMSCartItem *result = CartItemMapper.toModel(dictionary);
  
  XCTAssertNotNil(result, @"Result should not be nil");
  XCTAssertEqualObjects(result.itemId, @"testId", @"itemId should match");
  XCTAssertEqual(result.price, 12.34, @"price should match");
  XCTAssertEqual(result.quantity, 56.78, @"quantity should match");
}

- (void)test_toModel_nil {
  EMSCartItem *result = CartItemMapper.toModel(nil);
  
  XCTAssertNil(result, @"Result should be nil");
}

@end

File: android/src/test/java/com/emarsys/reactnative/utils/mappers/CartItemMapperTest.kt
package com.emarsys.reactnative.utils.mappers

import com.facebook.react.bridge.JavaOnlyMap
import org.junit.Assert
import org.junit.Test

class CartItemMapperTest {

  @Test
  fun test_toModel() {
    val map = JavaOnlyMap.of(
      "itemId", "testId",
      "price", 12.34,
      "quantity", 56.78
    )

    val result = CartItemMapper.toModel(map)

    Assert.assertNotNull(result)
    Assert.assertEquals(result!!.itemId, "testId")
    Assert.assertEquals(result!!.price, 12.34, 0.1)
    Assert.assertEquals(result!!.quantity, 56.78, 0.1)
  }

  @Test
  fun test_toModel_null() {
    val result = CartItemMapper.toModel(null)

    Assert.assertNull(result)
  }

}
